app/controllers/publishers/ats_api/v1/update_vacancy_service.rb

```diff
@@ -0,0 +1,63 @@
+module Publishers
+  module AtsApi
+    module V1
+      class UpdateVacancyService
+        def initialize(vacancy, params)
+          @vacancy = vacancy
+          @params = params
+        end
+
+        def call
+          if vacancy.update(permitted_params)
+            success_response
+          else
+            validation_error_response
+          end
+        end
+
+        private
+
+        attr_reader :vacancy, :params
+
+        def permitted_params
+          organisations = fetch_organisations(params[:schools])
+          raise ActiveRecord::RecordNotFound, "No valid organisations found" if organisations.blank?
+
+          params[:publish_on] ||= Time.zone.today.to_s
```

Comment:
if this were done after the merge, we would be altering a copy rather than the original?